akka-jackson/src/main/scala/akka/jackson/JacksonObjectMapperProvider.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.jackson
+
+import scala.util.Failure
+import scala.util.Success
+
+import akka.actor.DynamicAccess
+import akka.actor.ExtendedActorSystem
+import akka.actor.Extension
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.event.LoggingAdapter
+import com.fasterxml.jackson.annotation.JsonAutoDetect
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.PropertyAccessor
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.Module
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.paramnames.ParameterNamesModule
+import com.typesafe.config.Config
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object JacksonObjectMapperProvider {
+
+  /**
+   * Creates Jackson `ObjectMapper` with sensible defaults and modules configured
+   * in `akka.jackson.jackson-modules`.
+   */
+  def create(system: ExtendedActorSystem, jsonFactory: Option[JsonFactory]): ObjectMapper =
+    create(
+      system.settings.config,
+      system.dynamicAccess,
+      Some(Logging.getLogger(system, JacksonObjectMapperProvider.getClass)),
+      jsonFactory)
+
+  /**
+   * Creates Jackson `ObjectMapper` with sensible defaults and modules configured
+   * in `akka.jackson.jackson-modules`.
+   */
+  def create(
+    config:        Config,
+    dynamicAccess: DynamicAccess,
+    log:           Option[LoggingAdapter],
+    jsonFactory:   Option[JsonFactory]): ObjectMapper = {
+    import scala.collection.JavaConverters._
+
+    val mapper = new ObjectMapper(jsonFactory.orNull)
+
+    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+    mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY)
```

Comment:
It would be good to provide some way users can set additional configuration for the mapper. Otherwise, if they want to use any non-default configuration, they're stuck having to clone the whole apparatus and modify (a problem some Lagom users have run into).

Reply:
ok, then we need another configurable "provider" of the ObjectMapper

Reply:
ActorSystemSetup might be a solution for this